lib/components/viewers/stop-viewer.js

```diff
@@ -423,19 +424,23 @@ function getFormattedStopTime (stopTime, homeTimezone, soonText = 'Due', timeFor
 
   const now = moment().tz(homeTimezone)
   const serviceDay = moment(stopTime.serviceDay * 1000).tz(homeTimezone)
-  const currentHomeSecondsAfterMidnight = getHomeSecondsAfterMidnight(homeTimezone)
-  const differentDay = now.date() !== serviceDay.date()
-
+  const arrivesAfterMidnight = stopTime.realtimeDeparture >= ONE_DAY_IN_SECONDS
+  // Calculate current home seconds after midnight. If arrival occurs after
+  // midnight, adjust the value by one day in seconds.
+  const currentHomeSecondsAfterMidnight = getHomeSecondsAfterMidnight(homeTimezone) + (arrivesAfterMidnight ? ONE_DAY_IN_SECONDS : 0)
+  // Determine if arrival occurs on different day, making sure to account for an
+  // extra day added to the service day if it arrives after midnight.
+  const arrivalDay = arrivesAfterMidnight
+    ? serviceDay.add(1, 'day')
```

Comment:
Won't work for stop times that occur more than 1 day after the service date. Recommended change:

```javascript
serviceDay.add(
  (stopTime.realtimeDeparture - (stopTime.realtimeDeparture % ONE_DAY_IN_SECONDS)) / ONE_DAY_IN_SECONDS, 
  'day'
)

Reply:
How could a stop time occur more than 1 day after the service date? Are you suggesting that a stop time could be `49:00:00` (i.e., 1am two days after the service date)? Is that even permitted in GTFS? If so, this is such an edge case that I really don't think it's worth handling. Do we even know how OTP would handle such a stop time?

Reply:
OK, I guess I'll eat my words: https://groups.google.com/d/msg/transit-developers/ZkfnuNv1gho/ic6ObK5PBQAJ

>I haven’t analyzed existing feeds to look for patterns in when overnight service breaks happen, but as someone who has worked on several open source GTFS consuming apps I can say we’ve always considered the fact that some trains run for several days, and would at least accept and attempt to properly handle times up to 48:00. - Andrew Byrd